### DISCUSSION TITLE
CMake build help with TBB

### DISCUSSION THREAD

**Discussion by eddiehoyle** [Autodesk/maya-usd#1703] [Q&A] [answered]
This is more of a plea for help rather than a discussion, any help would be greatly appreciated.

I'm having trouble building [v0.10.0](https://github.com/Autodesk/maya-usd/tree/v0.10.0) against `usd-0.20.11` thanks to some very frustrating CMake warnings:

```
CMake Warning at plugin/adsk/plugin/CMakeLists.txt:18 (add_library):
  Cannot generate a safe linker search path for target mayaUsdPlugin because
  there is a cycle in the constraint graph:

    dir 0 is [/sw/packages/external/maya/2020.4.1627/platform-linux/arch-x86_64/os-CentOS-7/usr/autodesk/maya2020/lib]
      dir 1 must precede it due to link library [libtbb.so]
    dir 1 is [/sw/packages/external/tbb/2017.6/platform-linux/arch-x86_64/os-CentOS-7/lib/release]
      dir 0 must precede it due to link library [libtbb.so]
```

This is just _one_ of the dozens of warnings like this I'm seeing during my build.

For context, our usd build links against tbb-2017.6 (seen above). During maya-usd configure step, [FindMaya.cmake](https://github.com/Autodesk/maya-usd/blob/v0.10.0/cmake/modules/FindMaya.cmake#L213-L217) goes looking for tbb/tbb_debug libraries inside Maya's folders, finds them, but not without displaying the above warning for each `add_library()` defined in maya-usd's CMakeLists. No amount of `CMAKE_LIBRARY_PATH`, `LDFLAGS`, etc, settings has let me get around this. The build eventually passes and works correctly. However, I'm not exactly filled with confidence that it's safe for production use.

**My question**: I'm currently building usd against tbb-2017.6 as a standalone build. Should I be _also_ producing Maya variants of usd that link against Maya's tbb for use with maya-usd?

**Comment by seando-adsk** [comment, marked as answer]
Hi Eddie, we also have these cycle warnings on our internal builds at Autodesk. Our plugin builds are tested by QA and work correctly even though we have these warnings. They are something that we have been meaning to investigate.

Internally when building USD we actually get and use TBB and OSD from Maya, rather than letting USD download and build them as part of its build process. 

Sean

**Comment by eddiehoyle** [comment]
That's a relief! I'll continue to test on my end but now with added confidence from your QA team. Thanks, and appreciate the insight.
